src/tiled/objectselectiontool.cpp

```diff
@@ -687,6 +687,8 @@ static bool canResizeAbsolute(const MapObject *object)
     return false;
 }
 
+bool highlightState = false;
```

Comment:
This variable would really need to be a member of `ObjectSelectionTool`, otherwise you'll run into problems once two instances of the tool are being used (which is actually the case, since another instance of the tool is used in the tile collision editor).